### PR TITLE
feat: provide authorized miner mode

### PR DESCRIPTION
**Pull request by moreal**
It provides `--is-authorized-miner` flag for *authorized miner* mode. The mode makes miner mine only blocks that authorized miner should mine.

### CODE REVIEW

**Review by limebell — APPROVED** [review]

File: NineChronicles.Headless.Executable/Program.cs [hunk 1]
```diff
@@ -118,7 +118,11 @@ static void ConfigureSentryOptions(SentryOptions o)
             [Option(Description = "The secret key for AWS CloudWatch logging.")]
             string awsSecretKey = null,
             [Option(Description = "The AWS region for AWS CloudWatch (e.g., us-east-1, ap-northeast-2).")]
-            string awsRegion = null
+            string awsRegion = null,
+            [Option(Description =
+                "Flag to use as authorized miner. " +
+                "If true, it will mine only blocks that authorized miners should mine.")]
```
Comment (dahlia) [inline NineChronicles.Headless.Executable/Program.cs:124]: How about this?

> Run as an authorized miner, which mines only blocks that should be authorized.

File: README.md [hunk 2]
```diff
@@ -39,11 +38,12 @@ Options:
...
   --dev.reorg-interval <Int32>                             The size of reorg interval. Works only when dev mode is on.  0 by default. (Default: 0)
-  --log-minimum-level <String>                             The log minimum level during standalone execution. (Default: debug)
+  --log-minimum-level <String>                             The log minimum level during headless execution.  debug by default. (Default: debug)
   --aws-cognito-identity <String>                          The Cognito identity for AWS CloudWatch logging. (Default: )
   --aws-access-key <String>                                The access key for AWS CloudWatch logging. (Default: )
   --aws-secret-key <String>                                The secret key for AWS CloudWatch logging. (Default: )
   --aws-region <String>                                    The AWS region for AWS CloudWatch (e.g., us-east-1, ap-northeast-2). (Default: )
+  --is-authorized-miner                                    Flag to use as authorized miner. If true, it will mine only blocks that authorized miners should mine.
```
Comment (dahlia) [inline README.md:46]: I suggest to remove the `is-` prefix.  (Imagine what if `--strict-rendering` was `--is-strict-rendering`…)

File: NineChronicles.Headless.Executable/Program.cs (hunk 1 above)
Comment (moreal) [reply NineChronicles.Headless.Executable/Program.cs]: Amended in fba6e14

File: README.md (hunk 2 above)
Comment (moreal) [reply README.md]: Amended in 8744651

**Review by dahlia — APPROVED** [review]

**Pull request merged**